[Bug]: Legacy transactions fail on Ethereum and BNB with STX enabled

### Describe the bug

When STX are enabled Legacy transactions fail on mainnet and BNB with errors.

### Expected behavior

Legacy txs confirmed when STX are enabled

### Screenshots/Recordings

https://github.com/user-attachments/assets/fcb2cd9f-f595-49bc-98e3-4bee81a2ab64

### Steps to reproduce

1. Enable STX in the Settings. -> Advanced
2. Connect to the[ test dapp](https://metamask.github.io/test-dapp/)
3. Start Legacy transaction

### Error messages or log output

```shell
sentry-install.js:1 Error in smart transaction publish hook (Error#10)

sentry-install.js:1 Error#10: Fetch error:{"status":500,"statusCode":500,"message":"{\"reason\":\"processing response error\",\"code\":\"SERVER_ERROR\",\"body\":\"{\\\"jsonrpc\\\":\\\"2.0\\\",\\\"error\\\":{\\\"code\\\":-32000,\\\"message\\\":\\\"transaction at index 0 - is not a type 2 transaction (ie eip1559)\\\"},\\\"id\\\":42}\",\"error\":{\"code\":-32000},\"requestBody\":\"{\\\"method\\\":\\\"eth_sendSmartTransaction\\\",\\\"params\\\":[{\\\"transactions\\\":[{\\\"tx\\\":\\\"0xf8668202788502540be400825208940c54fccd2e384b4bb6f2e405bf5cbc15a017aafb808026a0ee448de6d3b4979801d4c31aa92a568075bbb5c91922eeff8ac82fa36bb7c986a01e53eca7a20b35eb74b1c0e491455f3faff36ab0a05e15a65d39e3e4319148a8\\\"}],\\\"auctionPeriod\\\":2}],\\\"id\\\":42,\\\"jsonrpc\\\":\\\"2.0\\\"}\",\"requestMethod\":\"POST\",\"url\":\"http://tx-sentinel-server-ethereum-mainnet.tx-sentinel/\"}"}
sentry-install.js:1   at chrome-extension://kddkfmaejcpdieoiemfcjenhmhoblgpo/background-2.js:1:609910
  at async V.submitSignedTransactions (chrome-extension://kddkfmaejcpdieoiemfcjenhmhoblgpo/background-2.js:1:596424)
  at async _.R (chrome-extension://kddkfmaejcpdieoiemfcjenhmhoblgpo/background-1.js:1:32669)
  at async _.submit (chrome-extension://kddkfmaejcpdieoiemfcjenhmhoblgpo/background-1.js:1:30545)
  at async chrome-extension://kddkfmaejcpdieoiemfcjenhmhoblgpo/common-3.js:5:798326
  at async Ue.approveTransaction (chrome-extension://kddkfmaejcpdieoiemfcjenhmhoblgpo/common-3.js:5:798231)
  at async Ue.processApproval (chrome-extension://kddkfmaejcpdieoiemfcjenhmhoblgpo/common-3.js:5:795871)
  at async chrome-extension://kddkfmaejcpdieoiemfcjenhmhoblgpo/background-1.js:1:62166
  at async chrome-extension://kddkfmaejcpdieoiemfcjenhmhoblgpo/background-1.js:1:37633
  at async chrome-extension://kddkfmaejcpdieoiemfcjenhmhoblgpo/common
```

### Detection stage

During release testing

### Version

12.15.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

--- a/app/scripts/lib/transaction/smart-transactions.test.ts
+++ b/app/scripts/lib/transaction/smart-transactions.test.ts
@@ -220,6 +220,22 @@ describe('submitSmartTransactionHook', () => {
     });
   });
 
+  it('falls back to regular transaction submit if it is a legacy transaction', async () => {
+    withRequest(async ({ request }) => {
+      // Modify transaction to be a legacy transaction (has gasPrice, no maxFeePerGas/maxPriorityFeePerGas)
+      request.transactionMeta.txParams = {
+        ...request.transactionMeta.txParams,
+        gasPrice: '0x77359400',
+      };
+      // Remove EIP-1559 specific fields if they exist
+      delete request.transactionMeta.txParams.maxFeePerGas;
+      delete request.transactionMeta.txParams.maxPriorityFeePerGas;
+
+      const result = await submitSmartTransactionHook(request);
+      expect(result).toEqual({ transactionHash: undefined });
+    });
+  });
+
   it('falls back to regular transaction submit if /getFees throws an error', async () => {
     withRequest(async ({ request, endFlowSpy }) => {
       jest

--- a/app/scripts/lib/transaction/smart-transactions.ts
+++ b/app/scripts/lib/transaction/smart-transactions.ts
@@ -25,6 +25,7 @@ import {
 
 import { decimalToHex } from '../../../../shared/modules/conversion.utils';
 import { CANCEL_GAS_LIMIT_DEC } from '../../../../shared/constants/smartTransactions';
+import { isLegacyTransaction } from '../../../../shared/modules/transaction.utils';
 import {
   SMART_TRANSACTION_CONFIRMATION_TYPES,
   ORIGIN_METAMASK,
@@ -143,7 +144,8 @@ class SmartTransactionHook {
     const useRegularTransactionSubmit = { transactionHash: undefined };
     if (
       !this.#isSmartTransaction ||
-      isUnsupportedTransactionTypeForSmartTransaction
+      isUnsupportedTransactionTypeForSmartTransaction ||
+      isLegacyTransaction(this.#transactionMeta)
     ) {
       return useRegularTransactionSubmit;
     }

